### DISCUSSION TITLE
Virtual Fields

### DISCUSSION THREAD

**Discussion by ghostlexly** [prisma/prisma#19835] [Q&A] [answered]
I would like to add some virtual fields on my entity.
When i use $extends to extends the PrismaClient, i can't add middlewares after to this client. The $use function doesn't exist.

```tsx
import { Prisma, PrismaClient } from "@prisma/client";

const prisma = new PrismaClient().$extends(uriExtension);

prisma.$use(); // this $use function doesn't exists
```

Is there a better way to add virtual fields on Prisma ?
Even if i can use $extends, it's looks so ugly when i use it for multiple times like in 5 separate purposes for 5 different fields.

**Comment by ludralph** [comment]
Hi @ghostlexly 👋 

Thank you for raising this question. The `$use` function is used when creating [middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware) with Prisma. To add virtual fields, you can take a look at this [example](https://www.prisma.io/blog/client-extensions-preview-8t3w27xkrxxn#example-computed-fields) of using Prisma Client Extension to add virtual fields.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by SevInf** [comment, marked as answer]
Hi @ghostlexly 
To use middleware with extensions you need to add all your middleware before you add any extension:

```ts
const _prisma = new PrismaClient()
_prisma.$use(middleware)
const prisma = _prisma.$extends(uriExtension)
```
